src/rpc/core_rpc_server.cpp

```diff
@@ -1296,6 +1299,7 @@ namespace cryptonote
       MERROR("Unknown invoke_http_mode: " << mode);
       return false;
     }
+    m_was_bootstrap_ever_used = true;
```

Comment:
You need to lock the mutex here too as you're writing to `m_was_bootstrap_ever_used`.